tests/select.spec.ts

```diff
@@ -1,21 +1,209 @@
 import { test, expect } from '@playwright/test';
+import { waitForCustomEvent } from './lib/helper.ts';
 
-test('user is able to make a selection with mouse', async ({ page }) => {
-  await page.goto('/test.html');
+test.beforeEach(async ({page}) => {
+  await page.goto('/select-test.html')
+})
 
-  await expect(page.getByLabel(/bobbins/i)).toBeVisible()
+test('Given an options attribute, on select the options should be visible and clicking an option should update the input', async ({ page }) => {
+  const select = page.getByTestId('basic-select')
+  await expect(select).toBeVisible()
 
-  const input = page.getByPlaceholder(/select a bobbins/i).last()
+  // don't expect the options to be rendered until a click on input
+  const optionsContainer = select.locator('.options-container').first()
+  await expect(optionsContainer).not.toBeVisible()
 
-  input.focus()
-  input.type('bogs')
+  // click on the input
+  const input = select.locator('input').first()
+  await input.focus()
+  await expect(optionsContainer).toBeVisible()
 
-  const option = page.getByText(/sticky bogs/i)
+  // within the optionsContainer there should be 3 different options
+  expect(await optionsContainer.locator('label').all()).toHaveLength(3)
 
-  await expect(option).toBeVisible()
+  await expect(optionsContainer).toHaveText(/one/)
+  await expect(optionsContainer).toHaveText(/two/)
+  await expect(optionsContainer).toHaveText(/three/)
 
-  option.click()
+  // selecting a value should emit an input event + render that as the input
+  const oneSelected = waitForCustomEvent(page, 'input')
```

Comment:
i juust merged in a helper function to also test for the any dispatched values -- would be good to update this to check for the value passed with the dispatched event